setup.php

```diff
@@ -81,51 +81,54 @@ function hashpass($pass, $salt = "") {
+                \$ppass   = '{$ppass}';
+                \$hp      = "{$hp}";
+                \$regtype = "{$_POST['reg']}";
+                \$path    = "{$_POST['path']}";
+                \$fpass   = {$_POST['fpass']};
+                \$li_shorten_only = {$_POST['li_shorten_only']};
+                \$theme   = "{$_POST['t']}";
+                \$ip      = {$_POST['ipfetch']};
+                \$li_show_front = {$_POST['li_show_front']};
+                \$unstr   = "{$rstr}";
+PHP;
 
 			    if (strlen($_POST['smtp-servers'])>1) {
-                    $smtpSection = '
-                        $smtpCfg = array(
-                            "servers"  => \''.$_POST['smtp-servers'].'\',
-                            "from" => \''.$_POST['smtp-from'].'\',
-                            "username" => \''.$_POST['smtp-username'].'\',
-                            "password" => \''.$_POST['smtp-password'].'\',
-                        );
-                    ';
+                    $smtpSection = <<<PHP
+                \n
+                \$smtpCfg = array(
+                    "servers"  => "{$_POST['smtp-servers']}",
+                    "from"     => "{$_POST['smtp-from']}",
+                    "username" => "{$_POST['smtp-username']}",
+                    "password" => "{$_POST['smtp-password']}",
+                );
+PHP;
                     $data .= $smtpSection;
                 }
-                $data .= '?>';
-                $file = "config.php";
-
-                $handle = fopen($file, 'a');
+                $data .= "\n?>";
+                $handle = fopen("config.php", 'w');
                 if (fwrite($handle, $data) === FALSE) {
-                    echo "Can not write to (" . $file . "). Please make sure your file permissions are correct.
-                    	<br />Your webserver's user should have write permissions for the folder. Try looking up <code>chown</code>, <code>chgrp</code> and <code>chmod</code>. <a href='http://linuxcommand.org/lts0070.php'>Helpful link</a>";
-                    die();
+                    die("<p class='alert alert-danger'>Could not write to <strong>config.php</strong>. Please make sure your file permissions are correct.
+                    	<br />Your webserver's user should have write permissions for the folder. Try looking up <code>chown</code>, <code>chgrp</code> and <code>chmod</code>. <a href='http://linuxcommand.org/lts0070.php'>Helpful link</a></p>. <strong>Aborting Setup</strong>");
+                } else {
+                    echo "<p class='alert alert-success'>Successfully created <strong>config.php</strong> .</p>";
                 }
-                echo "Successfully created config. ";
-                fclose($handle);
+                fclose($handle); //config.php
+
                 require_once('lib-core.php');
                 $path = $_POST['path'];
-                if (strlen($path) > 2) {
+                if (strlen($path) > 2) { // why 2? Is a folder like /r possible?
```

Comment:
It is 2 because we require the ending delimiter ("/r/" is allowed, "/r" is not a valid path for us). This has to do with apache's mod_rewrite path requirements for rewriting the base